EIPS/eip-6353.md

```diff
@@ -0,0 +1,333 @@
+
+    /**
+     * @dev Emitted when `toRemove` charity address is deleted from `whitelistedRate`.
+     */
+    event RemovedFromWhitelist (address toRemove);
+
+    /**
+     * @dev Emitted when `_defaultAddress` charity address is modified and set to `whitelistedAddr`.
+     */
+    event DonnationAddressChanged (address whitelistedAddr);
+
+    /**
+     * @dev Emitted when `_defaultAddress` charity address is modified and set to `whitelistedAddr` 
+    * and _donation is set to `rate`.
+     */
+    event DonnationAddressAndRateChanged (address whitelistedAddr,uint256 rate);
+
+    /**
+     * @dev Emitted when `whitelistedRate` for `whitelistedAddr` is modified and set to `rate`.
+     */
+    event ModifiedCharityRate(address whitelistedAddr,uint256 rate);
+    
+    /**
+    *@notice Called with the charity address to determine if the contract whitelisted the address
+    *and if it is the rate assigned.
+    *@param addr - the Charity address queried for donnation information.
+    *@return whitelisted - true if the contract whitelisted the address to receive donnation
+    *@return defaultRate - the rate defined by the contract owner by default , the minimum rate allowed different from 0
+    */
+    function charityInfo(
+        address addr
+    ) external view returns (
+        bool whitelisted,
+        uint256 defaultRate
+    );
+
+    /**
+    *@notice Add address to whitelist and set rate to the default rate.
+    * @dev Requirements:
+     *
+     * - `toAdd` cannot be the zero address.
+     *
+     * @param toAdd The address to whitelist.
+     */
+    function addToWhitelist(address toAdd) external;
+
+    /**
+    *@notice Remove the address from the whitelist and set rate to the default rate.
+    * @dev Requirements:
+     *
+     * - `toRemove` cannot be the zero address.
+     *
+     * @param toRemove The address to remove from whitelist.
+     */
+    function deleteFromWhitelist(address toRemove) external;
+
+    /**
+    *@notice Get all registered charity addresses.
+     */
+    function getAllWhitelistedAddresses() external ;
+
+    /**
+    *@notice Display for a user the rate of the default charity address that will receive donation.
+     */
+    function getRate() external view returns (uint256);
+
+    /**
+    *@notice Set personlised rate for charity address in {whitelistedRate}.
+    * @dev Requirements:
+     *
+     * - `whitelistedAddr` cannot be the zero address.
+     * - `rate` cannot be inferior to the default rate.
+     *
+     * @param whitelistedAddr The address to set as default.
+     * @param rate The personalised rate for donation.
+     */
+    function setSpecificRate(address whitelistedAddr , uint256 rate) external;
+
+    /**
+    *@notice Set for a user a default charity address that will receive donation. 
+    * The default rate specified in {whitelistedRate} will be applied.
+    * @dev Requirements:
+     *
+     * - `whitelistedAddr` cannot be the zero address.
```

Comment:
Why? If someone wants to burn money, they should be allowed :rofl:

Reply:
Why would they want to do that? 😭 🤣 Should I let them the choice?

Reply:
It saves a bit of gas to not check, and yeah, why not? You should try to write the standard as generally as possible. There may be a use case down the road where sending to the zero address is beneficial. Implementers could still prevent setting the zero address if they want.